### PR TITLE
20164-updated-alternate-names-table

### PR DESCRIPTION
**Pull request by jamespaologarcia**
*Issue #:* /bcgov/entity#20164

- Added entity_type to alternate_names & alternate_names1 tables to [alembic script](https://github.com/bcgov/lear/blob/2c24dbfce1c37bc78d1af75cdace2c829a5c8ee0/legal-api/migrations/versions/004f7558063f_second.py#L940).
- Added an index for entity_type field
- Ensured downgrade script is added too

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Review by chenhongjing — CHANGES_REQUESTED** [review]
Can you do the same things for `alternate_names_history` table?

**Comment by jamespaologarcia** [comment]
Making some updates. To address errors when running the upgrade.

File: legal-api/migrations/versions/004f7558063f_second.py [hunk 1]
```diff
@@ -964,11 +965,13 @@ def upgrade():
         batch_op.create_foreign_key(None, 'addresses', ['mailing_address_id'], ['id'])
         batch_op.create_foreign_key(None, 'filings', ['change_filing_id'], ['id'])
         batch_op.create_foreign_key(None, 'legal_entities', ['legal_entity_id'], ['id'])
+        batch_op.create_foreign_key('fk_entity_type', 'legal_entities', ['entity_type'], ['entity_type'])
```
Comment (argush3) [inline legal-api/migrations/versions/004f7558063f_second.py:968]: don't need a FK for `entity_type`

File: legal-api/migrations/versions/004f7558063f_second.py [hunk 2]
```diff
@@ -1065,6 +1068,7 @@ def downgrade():
         batch_op.drop_column('last_modified')
 
     with op.batch_alter_table('alternate_names', schema=None) as batch_op:
+        batch_op.drop_constraint('fk_entity_type', type_='foreignkey')
```
Comment (argush3) [inline legal-api/migrations/versions/004f7558063f_second.py:1071]: don't need this FK

**Comment by jamespaologarcia** [comment]
image

**Review by JazzarKarim — APPROVED** [review]
LGTM

**Review by chenhongjing — APPROVED** [review]
LGTM

**Pull request merged**